Framework/API/src/InfoComponentVisitor.cpp

```diff
@@ -39,6 +40,8 @@ void InfoComponentVisitor::registerComponentAssembly(
 
   m_ranges.emplace_back(std::make_pair(detectorStart, detectorStop));
 
+  // Record the ID -> index mapping
+  m_componentIdToIndexMap[assembly.getComponentID()] = m_componentIds.size();
```

Comment:
It won't be. Doing it in the visitor removes the need for the `ComponentInfo` to have to return the map. Not complete yet. I still have a lot of changes to push.